### PR TITLE
more robust parsing of JSON (+ indentation)

### PR DESCRIPTION
**Pull request by fsboehme**
- makes recursive calls to API (+ extra prompt about json parsability) if response was not parsable
- adds note about indentation to prompt.txt
- pass prompt.txt as system prompt
- uses env var for `DEFAULT_MODEL`
- uses env var for `OPENAI_API_KEY`

### CODE REVIEW

File: prompt.txt [hunk 1]
```diff
@@ -4,10 +4,13 @@ Because you are part of an automated system, the format you respond in is very s
...
 
+Be ABSOLUTELY SURE to include the CORRECT INDENTATION when making replacements.
+
+example response:
 [
   {"explanation": "this is just an example, this would usually be a brief explanation of what went wrong"},
   {"operation": "InsertAfter", "line": 10, "content": "x = 1\ny = 2\nz = x * y"},
   {"operation": "Delete", "line": 15, "content": ""},
-  {"operation": "Replace", "line": 18, "content": "x += 1"},
+  {"operation": "Replace", "line": 18, "content": "        x += 1"},
```
Comment (biobootloader) [inline prompt.txt:14]: Nice! Any idea how much better this makes it at indentation errors?

File: wolverine.py [hunk 2]
```diff
@@ -25,7 +33,41 @@ def run_script(script_name, script_args):
     return result.decode("utf-8"), 0
 
 
-def send_error_to_gpt(file_path, args, error_message, model):
+def send_error_to_gpt(file_path, args, error_message, model=DEFAULT_MODEL):
+    def json_validated_response(model, messages):
```
Comment (biobootloader) [inline wolverine.py:37]: nice idea, I saw 3.5-turbo especially had issues with this. Can you extract `json_validated_response` to be a standalone function? Seems a bit long to be an inner function

**Review by biobootloader — COMMENTED** [review]
🚀

File: prompt.txt (hunk 1 above)
Comment (fsboehme) [reply prompt.txt:14]: not a real measurement, but it seemed make it work more often than not (with gpt3.5) whereas initially it always got this wrong iirc.

File: wolverine.py (hunk 2 above)
Comment (fsboehme) [reply wolverine.py:37]: sure!

File: README.md [hunk 3]
```diff
@@ -26,6 +28,10 @@ You can also run with other models, but be warned they may not adhere to the edi
 
     python wolverine.py --model=gpt-3.5-turbo buggy_script.py "subtract" 20 3
 
+If you want to use GPT-3.5 by default instead of GPT-4 add this to `.env`:
+
+    DEFAULT_MODEL=gpt-3.5-turbo
+
```
Comment (biobootloader) [inline README.md:34]: I've just merged another PR that also set up .env usage. Can you fix conflicts and add this DEFAULT_MODEL to the .env.sample?

**Comment by fsboehme** [comment]
sorry, got a bit messy, but should be good to go 🙈

**Comment by biobootloader** [comment]
love it!

**Pull request merged**